bundles/org.openhab.binding.somfycul/src/main/java/org/openhab/binding/somfycul/internal/SomfyCULHandler.java

```diff
@@ -0,0 +1,260 @@
+                p.store(fw, "Initialized fields");
+            }
+        } else {
+            try (FileReader fr = new FileReader(file)) {
+                p.load(fr);
+            }
+        }
+
+        this.properties = p;
+    }
+
+    private long computeNewAddress(File directory) throws IOException {
+        File[] files = directory.listFiles((d, name) -> name != null && name.endsWith(".properties"));
+        if (files == null) {
+            throw new IOException("Cannot list files in " + directory.getAbsolutePath());
+        }
+
+        long maxAddr = 0;
+        for (File f : files) {
+            if (f.equals(propertyFile))
+                continue;
+
+            Properties other = new Properties();
+            try (FileReader fr = new FileReader(f)) {
+                other.load(fr);
+                String addr = other.getProperty("address");
+                if (addr != null) {
+                    try {
+                        long val = Long.decode("0x" + addr);
+                        maxAddr = Math.max(maxAddr, val);
+                    } catch (NumberFormatException ignored) {
+                    }
+                }
+            }
+        }
+
+        return maxAddr + 1;
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
+        super.thingUpdated(thing);
+
+        this.propertyFile = null;
+        this.properties = null;
+
+        initialize();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        Properties p = properties;
+        File file = propertyFile;
+
+        if (p == null || file == null) {
+            logger.warn("Ignoring command — properties not yet loaded");
+            return;
+        }
+
+        SomfyCommand somfyCommand = null;
+
+        switch (channelUID.getId()) {
+            case POSITION:
+                if (command instanceof UpDownType upDownCommand) {
+                    switch (upDownCommand) {
+                        case UP -> somfyCommand = SomfyCommand.UP;
+                        case DOWN -> somfyCommand = SomfyCommand.DOWN;
+                    }
+                } else if (command instanceof StopMoveType stopMoveCommand) {
+                    if (stopMoveCommand == StopMoveType.STOP) {
+                        somfyCommand = SomfyCommand.MY;
+                    }
+                }
+                break;
+
+            case PROGRAM:
+                if (command instanceof OnOffType) {
+                    // Don't check for on/off - always trigger program mode
+                    somfyCommand = SomfyCommand.PROG;
+                }
+                break;
+        }
+
+        if (somfyCommand == null)
+            return;
+
+        Bridge bridge = getBridge();
+        if (bridge == null)
+            return;
+
+        ThingHandler handler = bridge.getHandler();
+        if (!(handler instanceof CULHandler cul))
+            return;
```

Comment:
also applies to other checks below, always use braces.
```suggestion
        if (!(handler instanceof CULHandler cul)) {
            return;
        }
```